spec/my_enumerables_spec.rb

```diff
@@ -0,0 +1,176 @@
+# spec/my_enumerables_spec.rb
+
+require './my_enumerables.rb'
+
+describe Enumerable do
+  let(:array) { [1, 2, 3, 4, 5, 6, 7, 8, 9, 10] }
+  let(:range) { (0..10) }
+  let(:country_codes_hash)  { {
+    'kenya' => 254,
+    'uganda' => 256,
+    'eritrea' => 291
+  }}
+
+  context '#my_each' do
+
+    it 'returns an enumerator if block is not given' do
+      expect(array.my_each).to be_an_instance_of(Enumerator)
+    end
+
+    it 'returns the original array once it is done with the block' do
+      expect(array.my_each { |x| x * 2 }).to eql(array)
+    end
+
+    it 'returns the original range once it is done with the block' do
+      expect(range.my_each { |x| x * 2 }).to eql(range)
+    end
+
+    it 'returns the original hash once it is done with the block' do
+      expect(country_codes_hash.my_each { |country| "country - #{country} " }).to eql(country_codes_hash)
+    end
+  end
```

Comment:
- [ ] You should ensure that the specs for **EACH** method are wrapped in **ONE** `describe` block. In the highlighted code, a `describe` block should be wrapped around the `#my_each` method, then inside, you can have the `context` block which defines the condition the method it tested in and the `it` block shows the behavior of the method at each condition. For example, you should have something similar to this;

 ```
describe '#my_each' do
  context 'when a block is given' do
    it 'returns an the original array' do
      ....
    end
    ....
  end
 ...
end
```

**Note: This DOES NOT ONLY apply to the highlighted code. You should make the changes for all test blocks in this file.**